src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -325,3 +333,76 @@ func (c controller) sendCompleteInstallation(isSuccess bool, errorInfo string) {
 	}
 	c.log.Infof("Done complete installation step")
 }
+
+// get pod logs,
+// write tar.gz to pipe in a routine
+// upload tar.gz from pipe to assisted service.
+// close read and write pipes
+func (c controller) uploadPodLogs(podName string, namespace string) error {
+	c.log.Infof("Uploading logs for %s in %s", podName, namespace)
+	podLogs, err := c.kc.GetPodLogs(namespace, podName, logsSinceInSec)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get logs of pod %s", podName)
+	}
+	if podLogs == "" {
+		return errors.Errorf("no logs of %s were found", podName)
+	}
+	pr, pw := io.Pipe()
+	defer pr.Close()
+
+	go func() {
+		defer pw.Close()
+		err = utils.AddStringToTarGz(pw, podLogs, fmt.Sprintf("%s.logs", podName))
```

Comment:
I don't understand why do you complicate yourself with go routines/pipe.. This operation is synchronous
1. Creating compressed file -> 2. uploading it

Reply:
cause i don't have file.
i am running it as stream. I don't want another object

Reply:
that's why i prefer to right it to buffer and to read it from the same buffer

Reply:
and i don't think it is complicated. It is the way pipe is used. Even standard way